site/components/pages/Pledge/index.tsx

```diff
@@ -108,36 +133,16 @@ export const Pledge: NextPage = () => {
                     })}
                   />
                 ) : (
-                  <ConnectModal
-                    errorMessage={t({
-                      message:
-                        "We had some trouble connecting. Please try again.",
-                      id: "connect_modal.error_message",
-                    })}
-                    torusText={t({
-                      message: "or continue with",
-                      id: "connectModal.continue",
-                    })}
-                    titles={{
-                      connect: t({
-                        id: "connect_modal.sign_in",
-                        message: "Sign In / Connect",
-                      }),
-                      loading: t({
-                        id: "connect_modal.connecting",
-                        message: "Connecting...",
-                      }),
-                      error: t({
-                        id: "connect_modal.error_title",
-                        message: "Connection Error",
-                      }),
-                    }}
-                    buttonText={t({
+                  <ButtonPrimary
+                    variant="blue"
+                    label={t({
                       message: "Create a pledge",
                       id: "pledges.home.hero.create",
                     })}
-                    buttonVariant="blue"
-                    onClose={handleCreatePledge}
+                    onClick={() => {
+                      handleCreatePledge();
+                      toggleModal && toggleModal();
```

Comment:
this is preferred syntax for future reference. However when you update the initial state and the type as I suggested above, then you can just invoke directly and dont need to check for existence.
```suggestion
                      toggleModal?.();
```

* [ ] after you update the types/defaults please ctrl+shift+F for all invocations of toggleModal and renderModal and remove the null checks, invoke them all directly

Reply:
updated. noted on the syntax.